code/modules/reagents/chemistry/holder.dm

```diff
@@ -817,27 +817,11 @@
 		owner = reagent.holder.my_atom
 
 	//SKYRAT EDIT ADDITION BEGIN - CUSTOMIZATION
-	if(ishuman(owner))
-		var/mob/living/carbon/human/H = owner
-		//Check if this mob's species is set and can process this type of reagent
-		var/can_process = FALSE
-		//If we somehow avoided getting a species or reagent_flags set, we'll assume we aren't meant to process ANY reagents
-		if(H.dna && H.dna.species.reagent_flags)
-			var/owner_flags = H.dna.species.reagent_flags
-			if((reagent.process_flags & REAGENT_SYNTHETIC) && (owner_flags & PROCESS_SYNTHETIC))		//SYNTHETIC-oriented reagents require PROCESS_SYNTHETIC
-				can_process = TRUE
-			if((reagent.process_flags & REAGENT_ORGANIC) && (owner_flags & PROCESS_ORGANIC))		//ORGANIC-oriented reagents require PROCESS_ORGANIC
-				can_process = TRUE
-
-		//If the mob can't process it, remove the reagent at it's normal rate without doing any addictions, overdoses, or on_mob_life() for the reagent
-		if(!can_process)
-			reagent.holder.remove_reagent(reagent.type, reagent.metabolization_rate)
-			return
-	//We'll assume that non-human mobs lack the ability to process synthetic-oriented reagents (adjust this if we need to change that assumption)
-	else
-		if(reagent.process_flags == REAGENT_SYNTHETIC)
-			reagent.holder.remove_reagent(reagent.type, reagent.metabolization_rate)
-			return
+	var/can_process = reagent_process_flags_valid(owner, reagent)
+	//If the mob can't process it, remove the reagent at it's normal rate without doing any addictions, overdoses, or on_mob_life() for the reagent
+	if(!can_process)
+		reagent.holder.remove_reagent(reagent.type, reagent.metabolization_rate)
+		return
```

Comment:
What's the reasoning behind this change, in relation to the rest of this PR?

Reply:
i just extracted the process flag logic to its own proc (was needed for the trauma code), if you look down in the removed code you'll see this was part of the original skyrat edit

if you're getting confused on the else if (reagent.process_flags == REAGENT_SYNTHETIC), it's still covered under this proc